src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -53,6 +55,17 @@ public partial class NodeViewModel : ViewModelBase
         public readonly WorkspaceViewModel WorkspaceViewModel;
         public readonly Size? PreferredSize;
 
+        private bool previewPinned;
+        public bool PreviewPinned {
+            get { return previewPinned; }
+            set
+            {
+                previewPinned = value;
```

Comment:
Here you can write next:
```c#
if(previewPinned == value) return;
previewPinned = value;
                 if (PreviewPinEvent != null)
                     PreviewPinEvent(previewPinned);
```

In order to not fire PreviewPinEvent, when pin value is the same.